docker_ci_deploy/__main__.py

```diff
@@ -9,15 +9,49 @@
 from itertools import chain
 
 
+# Reference regexes for parsing Docker image tags into separate parts.
+# https://github.com/docker/distribution/blob/v2.6.0-rc.2/reference/regexp.go
+HOSTNAME_COMPONENT_PATTERN = (
+    r'(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])')
+# hostname = hostcomponent ['.' hostcomponent]* [':' port-number]
+HOSTNAME_PATTERN = (
+    HOSTNAME_COMPONENT_PATTERN +
+    r'(?:(?:\.' + HOSTNAME_COMPONENT_PATTERN + r')+)?'
+    r'(?::[0-9]+)?')
```

Comment:
Would it make sense to use `.format()` here? We only want that if it'll be cleaner, though.